core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/filter_store.go

```diff
@@ -23,7 +23,9 @@ type UpkeepFilterStore interface {
 var _ UpkeepFilterStore = &upkeepFilterStore{}
 
 type upkeepFilter struct {
-	addr     []byte
+	addr []byte
+	// selector is the filter selector in log trigger config
+	selector uint8
```

Comment:
This will also need to be set during construction

Reply:
it looks like this is done in `RegisterFilter` in `provider_life_cycle.go` already